test/libyul/yulOptimizerTests/fullSuite/combine_shift_and_and_unsplit.yul

```diff
@@ -14,6 +14,6 @@
 // {
 //     {
 //         let x := calldataload(0)
-//         sstore(shr(248, x), and(shr(4, x), 3855))
+//         sstore(shr(248, x), and(and(shr(4, x), sub(shl(244, 1), 1)), 3855))
```

Comment:
After looking into it, I see it's just that it's not fully optimized here. `ExpressionSimplifier` (`s`) in the default sequence eventually figures out that part of this expression is an application of two constant masks (`(4 >> x) & 0x01e7 & 0xf0f`) and one of them can be safely dropped because only the three lowest bits matter and those are identical.

The simplification does not take place though if the expression is not split. In the default sequence that happens eventually due to repetition, in the new one it does not. I found that adding `ExpressionSplitter` (`x`) resolves this. Either just `x` before `[scCTUt]` or `xsu` somewhere later in the sequence.

Reply:
### Comparison with unmodified `the-good-parts-mk2` on `cancun`
#### Tweak
```diff
-       "[scCTUt] TOntnfDIul"          // Perform structural simplification
+       "x[scCTUt] TOntnfDIul"         // Perform structural simplification
```

#### External tests (`ir-optimize-evm+yul`)
[`seqbench-sequence-the-good-parts-mk2-tweaks-x-before-s`](https://app.circleci.com/pipelines/github/ethereum/solidity/33660/workflows/906dd0f1-b678-441a-be5c-06474ba0f3d3) vs [`seqbench-sequence-the-good-parts-mk2-cancun`](https://app.circleci.com/pipelines/github/ethereum/solidity/33655/workflows/f901ee55-094e-44b1-860c-44cab6b675bb)

|          project |  bytecode_size | deployment_gas |     method_gas |
|:----------------:|---------------:|---------------:|---------------:|
|            brink | **`+0.06% ❌`** |                |                |
|        elementfi |  **`-0.1% ✅`** |                |                |
|              ens | **`-0.26% ✅`** |           `0%` |          `+0%` |
|            euler | **`-0.13% ✅`** |                |                |
|              gp2 | **`-0.04% ✅`** |                |                |
|  perpetual-pools | **`-0.33% ✅`** | **`-0.33% ✅`** | **`-0.11% ✅`** |
|    pool-together | **`-0.06% ✅`** |                |                |
|          uniswap | **`-0.33% ✅`** |                |                |
| yield_liquidator | **`-0.07% ✅`** | **`-0.08% ✅`** |  **`+0.1% ❌`** |
|         zeppelin | **`-0.08% ✅`** | **`-0.07% ✅`** | **`+0.06% ❌`** |

#### Gas diff stats
| File name                                                       | IR optimized   | Legacy optimized   | Legacy   |
|-----------------------------------------------------------------|----------------|--------------------|----------|
| `externalContracts/base64.sol`                                             | -1%            |                    |          |
| `array/array_storage_push_pop.sol`                                         | -1%            |                    |          |
| `array/function_array_cross_calls.sol`                                     | -1%            |                    |          |
| `array/push/byte_array_push_transition.sol`                                | -1%            |                    |          |
| `array/pop/byte_array_pop_long_storage_empty.sol`                          | -1%            |                    |          |
| `array/byte_array_transitional_2.sol`                                      | -1%            |                    |          |